solana-kotlin/src/commonMain/kotlin/net/avianlabs/solana/SolanaClient.kt

```diff
@@ -9,10 +10,12 @@ public class SolanaClient(
   private val client: RpcKtorClient,
 ) {
 
+  @OptIn(ExperimentalSerializationApi::class)
   internal val json: Json = Json {
     ignoreUnknownKeys = true
     isLenient = true
     allowSpecialFloatingPointValues = true
+    explicitNulls = false
```

Comment:
what made it be required?

Reply:
I was only playing with this. I don't think we explicitly need this . But with all those fields being optional I'd rather NOT send any `param = null` as a precaution.

Reply:
then you end up doing things like 

```
commitment?.let { add(json.encodeToJsonElement(BalanceParams(it.value))) }
```

which will break the moment `BalanceParams` have more than 1 nullable (optional) field 

🤔

Reply:
hmmm I don't think those two things are related. 

This flag only doesn't add to the json null fields. so for example for `isBlockHashValid` call that has two optional params (commitment and minContextSlot) it would be : 
```
params: [
   block_hash_122, 
   {
      "commitment" : "confirmed"
   }
]
```
instead of : 
```
params: [
   block_hash_122, 
   {
      "commitment" : "confirmed"
       "minContextSlot" : null  
   }
]
```

Reply:
my bad! mixed them up. so even better, you'd always want to add `BalanceParams`?

Reply:
So at the moment there is 
`commitment?.let { add(json.encodeToJsonElement(BalanceParams(it.value))) }` and `commitment?.let` guards us to **not** add empty object. 

This flag is handy in  cases like `IsBlockHashValid()` where you have two nullable fields.
   - If both are null we don't add the object at all with an if statement. 
    - When one them is null we don't add the `null` value to json thanks to this flag. 
    
```
if (commitment != null || minContextSlot != null) {
   add(json.encodeToJsonElement(IsBlockHashValidParams(commitment?.value, minContextSlot)))
}
```

Reply:
that seems more error prone than... just add the object and the nulls, because the RPC doesn't seem to mind (or does it break? I don't remember it breaking down)

Reply:
I don't know because the only real use case we had in would always add some params.
The server _should_ read those nulls correctly, but I would rather not add empty objects like `{ }`

I think it's just being a good citizen and not send things when they are not needed.